src/helpers/parser.ts

```diff
@@ -25,19 +29,52 @@ export const gitIssueParser = async ({ owner, repo, issue_number }: GitParser):
   }
 };
 
-export const gitLinkedIssueParser = async ({ owner, repo, issue_number }: GitParser): Promise<string> => {
+export const gitLinkedIssueParser = async ({ owner, repo, pull_number }: GitParser): Promise<string> => {
   try {
-    const { data } = await axios.get(`https://github.com/${owner}/${repo}/pull/${issue_number}`);
+    const { data } = await axios.get(`https://github.com/${owner}/${repo}/pull/${pull_number}`);
     const dom = parse(data);
     const devForm = dom.querySelector("[data-target='create-branch.developmentForm']") as HTMLElement;
-    const linkedPRs = devForm.querySelectorAll(".my-1");
+    const linkedIssues = devForm.querySelectorAll(".my-1");
 
-    if (linkedPRs.length === 0) {
+    if (linkedIssues.length === 0) {
       return "";
     }
 
-    const prUrl = linkedPRs[0].querySelector("a")?.attrs?.href || "";
-    return prUrl;
+    const issueUrl = linkedIssues[0].querySelector("a")?.attrs?.href || "";
+    return issueUrl;
+  } catch (error) {
+    return "";
+  }
+};
+
+export const gitLinkedPrParser = async ({ owner, repo, issue_number }: GitParser) => {
+  try {
+    const { data } = await axios.get(`https://github.com/${owner}/${repo}/issues/${issue_number}`);
+    const context = getBotContext();
+    const payload = context.payload as Payload;
+    const dom = parse(data);
+    const devForm = dom.querySelector("[data-target='create-branch.developmentForm']") as HTMLElement;
+    const linkedPRs = devForm.querySelectorAll(".my-1");
+    if (linkedPRs.length === 0) return "";
+    let linkedPullRequest;
+    for (const linkedPr of linkedPRs) {
+      const prHref = linkedPr.querySelector("a")?.attrs?.href || "";
+      const parts = prHref.split("/");
+      // extract the organization name and repo name from the link:(e.g. "https://github.com/wannacfuture/ubiquibot/pull/5";)
+      const organization = parts[parts.length - 4];
+      const repository = parts[parts.length - 3];
+
+      if (`${organization}/${repository}` !== payload.repository.full_name) continue;
+      const prNumber = prHref.substring(prHref.lastIndexOf("/") + 1);
+      const pr = await getPullByNumber(context, Number(prNumber));
+      if (!pr || !pr.merged) continue;
+
+      if (!linkedPullRequest) linkedPullRequest = pr;
+      else if (linkedPullRequest.merged_at && pr.merged_at && new Date(linkedPullRequest.merged_at) < new Date(pr.merged_at)) {
+        linkedPullRequest = pr;
+      }
+    }
+    return linkedPullRequest;
```

Comment:
Here, check the organization and repo name(compare with the bot's org and repo).
Get the latest merged pull request.

Ready for review. @whilefoo , @pavlovcik

Reply:
Github search API is also a nice alternative if considered. It will not require HTML parsing and probably will not need timestamp checking. All hard work will shift to github. Here is an example of an issue mentioned by @whilefoo #583

https://api.github.com/search/issues?q=repo:ubiquity/ubiquibot+is:pr+is:merged+in:body%20Resolves%20%23583

Let's break down and  see what it is searching for:

```
repo:ubiquity/ubiquibot+
is:pr+
is:merged+
in:body Resolves #583
```
But it should be tried on other use cases too to be sure if it is working well.

Reply:
> https://api.github.com/search/issues?q=repo:ubiquity/ubiquibot+is:pr+is:merged+in:body%20Resolves%20%23583

That's great but if you link the PR manually within the UI then it won't work